restapi/src/test/java/gov/cdc/sdp/cbr/restapi/LogApiTest.java

```diff
@@ -0,0 +1,395 @@
+package gov.cdc.sdp.cbr.restapi;
+
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Message;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.impl.DefaultMessage;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultHandler;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import com.google.gson.Gson;
+
+import gov.cdc.sdp.cbr.HL7V2BatchSplitter;
+import gov.cdc.sdp.cbr.model.SDPMessage;
+import gov.cdc.sdp.cbr.trace.TraceService;
+import gov.cdc.sdp.cbr.trace.model.TraceLog;
+import gov.cdc.sdp.cbr.trace.model.TraceStatus;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = { "classpath:context.xml" })
+@WebAppConfiguration
+@TestPropertySource(properties = { "input.post.endpoint:direct:test-target" })
+
+@PropertySource("classpath:application.properties")
+public class LogApiTest {
```

Comment:
At some point, this test is trying to hit a table called `trace_log_api_test`. Can we have the set up create this table? I know that I can create it manually but when we move CBR to continuous integration, it won't work.